crates/publish/src/oci/client.rs

```diff
@@ -248,3 +311,94 @@ impl Client {
         }
     }
 }
+
+#[derive(Serialize, Deserialize)]
+struct AuthConfig {
+    /// Map between registry server and base64 encoded username:password credential set.
+    pub auths: HashMap<String, String>,
+}
+
+impl AuthConfig {
+    /// Load the authentication configuration from the default location
+    /// ($XDG_CONFIG_HOME/fermyon/registry-auth.json).
+    pub async fn load_default() -> Result<Self> {
```

Comment:
Do we want this path to be configurable?

I would really like us to refactor the way we handle configuration throughout Spin, and have a central place where we handle reading and writing configuration...

Reply:
Yeah I've been thinking about how to centralize this sort of thing. We could call it `spin-config` no wait... `spin-core`! no.... `spin-app`? shoot...